pkg/api/handlers/libpod/generate.go

```diff
@@ -32,11 +32,8 @@ func GenerateSystemd(w http.ResponseWriter, r *http.Request) {
 		After            []string `schema:"after"`
 		Requires         []string `schema:"requires"`
 	}{
-		StartTimeout:    0,
-		StopTimeout:     util.DefaultContainerConfig().Engine.StopTimeout,
-		ContainerPrefix: "container",
-		PodPrefix:       "pod",
-		Separator:       "-",
```

Comment:
I appreciate your review and help. Could you please expand on how making the fields a pointer would work? I am not able to understand this strategy. Thank you.

Reply:
Sorry this comment got lost in the inbox.
change this to
```
ContainerPrefix  *string   `schema:"containerPrefix"`
PodPrefix        *string   `schema:"podPrefix"`
Separator        *string   `schema:"separator"`
```

Leave the default unset.

Now in the code below you can check if var == nil, in this case you set the correct default of `container`, `pod`, `-`.
If not nil you can just use the value.

Note this is just my theory I have not tested if this will actually work. But this change will allow us to differentiate between an empty string ("") or not set at all (nil).

Reply:
I made this change and it works.